overlord/snapstate/snapstate.go

```diff
@@ -890,18 +899,34 @@ func canRemove(si *snap.Info, snapst *SnapState, removeAll bool) bool {
 	}
 
 	// TODO: use Required for these too
+
 	// Gadget snaps should not be removed as they are a key
 	// building block for Gadgets. Do not remove their last
 	// revision left.
 	if si.Type == snap.TypeGadget {
 		return false
 	}
 
-	// You never want to remove a kernel or OS Do not remove their
+	// FIXME: alternatively, pass state into canRemove() and check
+	//        if "core" and "ubuntu-core" are both installed.
+	//
+	// Allow "ubuntu-core" removals here because we might have two
+	// core snaps installed (ubuntu-core and core). Note that
+	// ideally we would only allow the removal of "ubuntu-core" if
+	// we know that "core" is installed too and if we are part of
+	// the "ubuntu-core->core" transition. But this transition
+	// starts automatically on startup so the window of a user
+	// triggering this manually is very small.
+	if si.Name() == "ubuntu-core" {
```

Comment:
I think it would be clearer though if there was si.Type == snap.TypeOS in the condition as well